packages/nextjs/services/database/config/schema.ts

```diff
@@ -50,19 +58,28 @@ type ExternalLink = Partial<{
 }>;
 
 export const challenges = pgTable("challenges", {
-  id: varchar({ length: 255 }).primaryKey(), // Unique identifier for the challenge
+  id: varchar({ length: 255 }).primaryKey(),
   challengeName: varchar({ length: 255 }).notNull(),
   description: text().notNull(),
   sortOrder: integer().notNull(),
-  github: varchar({ length: 255 }), // Repository reference for the challenge
-  autograding: boolean().default(false), // Whether the challenge supports automatic grading
+  github: varchar({ length: 255 }),
+  autograding: boolean().default(false),
   disabled: boolean().default(false),
   previewImage: varchar({ length: 255 }),
   icon: varchar({ length: 255 }),
   dependencies: varchar({ length: 255 }).notNull().array(),
   externalLink: jsonb("external_link").$type<ExternalLink>(),
 });
 
+export const batches = pgTable("batches", {
+  id: serial().primaryKey(),
+  name: varchar({ length: 255 }).notNull(),
```

Comment:
In BG3.5 we are saving it with random UUID as id and name as "1", "2", etc.

Here I'm using "serial" (since we are doing it in the userChallenges table) + name will be "Batch 1", etc (instead of just "1") so it allows some flexibility (if we do a batch for univeristy people, it might be call different). It'd be like a display name (and we can always sort by startDate or ID if serial)

What do you think?

Reply:
Sounds good to me!